Mathlib/Data/Finset/Basic.lean

```diff
@@ -807,6 +807,11 @@ theorem ssubset_iff_exists_cons_subset : s ⊂ t ↔ ∃ (a : _) (h : a ∉ s),
   obtain ⟨a, hs, ht⟩ := not_subset.1 h.2
   exact ⟨a, ht, cons_subset.2 ⟨hs, h.subset⟩⟩
 
+theorem cons_swap (ha : a ∉ s) (hb : b ∉ s.cons a ha) :
+    (s.cons a ha).cons b hb = (s.cons b fun h ↦ hb (mem_cons.mpr (.inr h))).cons a fun h ↦
+      hb (mem_cons.mpr (.inl ((mem_cons.mp h).elim symm (fun h ↦ False.elim (ha h))))) := by
+  apply Subset.antisymm <;> intro _ _ <;> simp [mem_cons] at * <;> tauto
```

Comment:
A slightly slicker proof:
```suggestion
      hb (mem_cons.mpr (.inl ((mem_cons.mp h).elim symm (fun h ↦ False.elim (ha h))))) := by
  Finset.coe_injective <| by simpa using Set.insert_comm b a s
```
this also raises that probably you have `a` and `b` backwards in the statement.

Reply:
I didn't change the order of `a` and `b` yet, as `b` being dependent on `a` is more likely than the other way around,

Reply:
I'm just suggesting you swap the names around, not change anything else about the assumptions.